Feat: create a like button

motuncoded opened the pull request:
In this change I created a like button that when clicked changes the number of likes it has.
closes #7

Commit:
Create a simple scroll of posts

In this feature, I used Posts Placeholder Api which  render a simple scroll of different posts.

Commit:
Setup the prettier and made file changes

In this change, I set up prettier in the project and move the function types outside the Reactjs function

Commit:
Create an infinite scroll of posts

In this change, I created a Posts.tsx file that can allow infinite scroll of posts with loading when navigatimg down the posts.

Commit:
Create an infinite scroll of posts

In this change, I created a Posts.tsx file that can allow infinite scroll of posts with loading when navigatimg down the posts.

motuncoded committed:
Merge branch 'main' into infinite-scroll-of-posts

Commit:
Fix the infinite scroll of posts

In this change I fixed the changes based on the review to have an infinite scroll of posts

Commit:
Merge branch 'infinite-scroll-of-posts' of https://github.com/motuncoded/social_website into infinite-scroll-of-posts

Commit:
optimising next build

In this change I optimised the project by runnning the nex build command to be able to check the vercel preview

Commit:
adjust the files

In this change, I removed the posts.tsx and incorporated the rendering of the API inside the index.tsx file to avoid rendering the same thing

Commit:
Fix the problem of the loading state

In this change, I made the loading state to render the right jsx elements for loading and no more posts

Commit:
Create a functional like button

In this change I created a like button that when clicked changes the number of likes it has

motuncoded committed:
Merge branch 'main' into like-button

Asylcreek commented on pages/index.tsx:
```diff
@@ -11,6 +14,8 @@ type Post = {
   title: string;
   body: string;
   userId: number;
+  likes: number;
+  liked: boolean;
```
I'm a little confused ... should you know if a post is liked by checking the `likes`? What's the `liked` for?

motuncoded replied:
the liked is a boolean if true icon IoHeart will shows if otherwise IoHeartOutline will show

Asylcreek replied:
Yeah, but since you have `users` in your app, you cannot have the `liked` on the post, it should account for the user who liked the post

Asylcreek commented on pages/index.tsx:
```diff
@@ -37,7 +42,11 @@ export default function Home() {
         setHasMore(true);
       } else if (posts.length === 0) {
         setHasMore(true);
```
Why?

motuncoded replied:
This is know that there is no more post at the end of the posts

Asylcreek replied:
But how does no post (posts.length===0) equate to there being more posts?

Commit:
fix the like button and loading state of the posts

In this change I removed the liked type and also change th loading state of the posts

Commit:
fix the like button and loading state of the posts

In this change I removed the liked type and also change th loading state of the posts

Commit:
fix the like button and loading state of the posts

In this change I removed the liked type and also change th loading state of the posts

Commit:
fix the like button and loading state of the posts

In this change I removed the liked type and also change th loading state of the posts

Commit:
fix the like button and loading state of the posts

In this change I removed the liked type and also change th loading state of the posts

Asylcreek commented on pages/index.tsx:
```diff
@@ -35,10 +38,8 @@ export default function Home() {
-      } else if (posts.length === 0) {
-        setHasMore(true);
       } else {
-        setHasMore(false);
+
       }
```
Shouldn't you have removed the else block

motuncoded replied:
Will check it...

Commit:
added the else statement of the fetchPosts

In this change I added the else statement in the ftchPosts
